src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/HoverComponents.test.ts

```diff
@@ -39,6 +39,32 @@ suite('Hover Components', () => {
         }, /* timeout */ 3000, /* pollInterval */ 500, true /* suppress timeout */);
     });
 
+    test('Can perform hovers on directive attributes', async () => {
+        // Hover over '@onclick'
+        const counterPath = path.join(componentRoot, 'Components', 'Pages', 'Counter.razor');
+        const counterDoc = await vscode.workspace.openTextDocument(counterPath);
+        const hoverResult = await vscode.commands.executeCommand<vscode.Hover[]>(
+            'vscode.executeHoverProvider',
+            counterDoc.uri,
+            new vscode.Position(6, 36));
+
+        assert.ok(hoverResult, 'Should have a hover result for @onclick');
+        if (!hoverResult) {
+            // Not possible, but strict TypeScript doesn't know about assert.ok above.
+            return;
+        }
+
+        assert.equal(hoverResult.length, 1, 'Something else may be providing hover results');
+
+        const onClickResult = hoverResult[0];
+        const expectedRange = new vscode.Range(
+            new vscode.Position(6, 31),
```

Comment:
If someone changes the `Components/Pages/Counter.razor` file in any way these positions will probably be wrong. I'd suggest inserting the element with `@onclick` at the top of the file and verifying it ther.